src/ignitus-Common-Auth-Student-Prof/Components/createSignInPage.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { withErrorBoundary } from '../../ignitus-Internals';
+import * as t from './constants';
+import '../Styles/style.scss';
+
+interface SignInConfig {
+  title: string;
+  studentRoute: string;
+  professorRoute: string;
+}
+
+function createSignInPage(config: SignInConfig): React.ComponentType {
```

Comment:
@steric85 LGTM , but 1 Question do you really think that the name of the component should be `createSignInPage` as semantically it means it is an component for `signIn/login.` I think it would be great if we can name it something like `commonLoginRegisterComponent` or only `commonLoginRegister` :) , let me know your thought on it.

Reply:
I agree with you. I wasn't able to think of a common name for login and sign up.  I'll update it to `commonLoginRegister`.

Reply:
Uhmm, on a second thought, as it is something that returns a component i.e. creates a component for login/register, should we name it `createCommonLoginRegister` ?